R/source.R

```diff
@@ -23,6 +28,7 @@ source_file <- function(path, env = test_env(), chdir = TRUE,
   con <- textConnection(lines, encoding = "UTF-8")
   on.exit(try(close(con), silent = TRUE), add = TRUE)
   exprs <- parse(con, n = -1, srcfile = srcfile, encoding = "UTF-8")
+  exprs <- filter_label(exprs, label, error_call = error_call)
```

Comment:
Does this preserve source references?

Reply:
I believe so because they're attached to the individual expressions. Worth double-checking though.